[libgit2] Update to 1.8.0

razielanarki opened the pull request:
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

razielanarki committed:
Update libgit2 to 1.8.0

Cheney-W commented on ports/libgit2/portfile.cmake:
```diff
@@ -1,9 +1,9 @@
-    REF v1.6.4
-    SHA512 fd73df91710f19b0d6c3765c37c7f529233196da91cf4d58028a8d3840244f11df44abafabd74a8ed1cbe4826d1afd6ff9f01316d183ace0924c65e7cf0eb8d5
-    HEAD_REF maint/v1.6
+    REF v1.8.0
```
```suggestion
    REF "v${VERSION}"
```

razielanarki replied:
i'm not 100% sure/convinced it's a good idea to "generalize" the portfile version into a "{VAR}" so atm i havent

Cheney-W commented:
I will be converting your PR to draft status. The above suggested changes are only recommendations. If you are willing to adopt them, please click "ready for review" after making the modifications. If you do not wish to make any changes, please click "ready for review" directly. That way, I can be aware that you've responded since you can't modify the tags.

razielanarki committed:
Update libgit2 to 1.8.0: update pkg features

razielanarki commented:
had some time adjust the package features a bit, which i wanted to do before _ready_-ing the submission:
- added `openssh` and `libssh2` to select between the new openssh exec based and the old ssh impl  
- also  there's `ssh` which selects the default (`libssh2`) impl
- added `schannel` selecting the 'Schannel' SSL backend (windows only)

razielanarki committed:
Update libgit2 to 1.8.0: typoes, fixes

dg0yt commented on ports/libgit2/portfile.cmake:
```diff
@@ -1,112 +1,126 @@
+    set(USE_HTTPS ${VALUE} PARENT_SCOPE)
+endfunction()
+
+function(set_ssh_backend VALUE)
+    if(USE_SSH)
+        message(FATAL_ERROR "Only one SSH backend (openssh,libssh2) is allowed")
+        message(FATAL_ERROR "SSH backend already set")
+    endif()
+    set(USE_SSH ${VALUE} PARENT_SCOPE)
+endfunction()
```
I don't think it is acceptable to add *another* choice of alternative backends. `libgit2` is already listed as an example of features implementing alternatives in a way which would not be accepted today (but are retained for compatibility).
https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#do-not-use-features-to-implement-alternatives

razielanarki replied:
Hmm, did not know that, thanks for pointing it out!

Can you provide your thoughts on my attempted resolution? 

Basically I reverted the `ssh` feature to the one in v1.6.4, and made `openssh` to be not mutually exclusive - on the portfile level that is, because libgit2 still implements the feature as a choice (using one or the other, but not both)

(As a last resort I may have to fall-back to using build-time messages as [outlined here](https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#recommended-solutions), but I'd like to avoid that if possible)

Cheney-W replied:
I don't think the `schannel` feature will be accepted because it violates the rule mentioned above.

Cheney-W committed:
Update libgit2.json

Cheney-W commented:
Please do not modify line breaks of the `portfile.cmake` file, as this will cause the entire file to be displayed on the `Files Changed` page.

razielanarki committed:
fixes as per review:

- attempt to make the `openssh` feature follow guidelines more closely
- revert accicendtal line ending change in the portfile

razielanarki committed:
fix ninja error

razielanarki committed:
alternate fix for ninja error

razielanarki committed:
fix: revert addition of `libssh2` feature

razielanarki commented:
i think i have everything in order now for another round of review.
thanks!

dg0yt commented on ports/libgit2/vcpkg.json:
```diff
@@ -30,6 +29,9 @@
       ]
     },
+    "openssh": {
+      "description": "SSH support (OpenSHH)"
```
```suggestion
      "description": "SSH support (OpenSSH)"
```

There is no dependency here. Is this using an external program (seeing the `exec` in the portfile`) instead of a lib?

razielanarki replied:
fixed the typo: replaced with a more clear feature descriuption

Cheney-W commented on ports/libgit2/portfile.cmake:
```diff
@@ -1,9 +1,9 @@
-    HEAD_REF maint/v1.6
+    REF v1.8.0
+    SHA512 e5634267bd9c6a594c9a954d09c657e7b8aadf213609bf7dd83b99863d0d0c7109a5277617dd508abc2da54ea3f12c2af1908d1aeb73c000e94056e2f3653144
+    HEAD_REF master
```
```suggestion
    HEAD_REF main
```
The default branch of the upstream is `main`.

razielanarki replied:
fixed

razielanarki committed:
fix typos, clarify `openssh` feature description

Cheney-W commented:
Add `reviewed` tag for further review by the team.

JavierMatosD commented on ports/libgit2/portfile.cmake:
```diff
@@ -35,7 +35,7 @@ endfunction()
 function(set_tls_backend VALUE)
     if(USE_HTTPS)
-        message(FATAL_ERROR "Only one TLS backend (openssl,winhttp,sectransp,mbedtls) is allowed")
+        message(FATAL_ERROR "Only one TLS backend (openssl,winhttp,schannel,sectransp,mbedtls) is allowed")
```
Adding features as alternatives is explicitly disallowed. The fact that this port already does so is unfortunate, but we can't add more.

razielanarki replied:
removed the schannel alternative (which - and I'm no expert at this - is just another way of using winhttp on enabled/newer systems)

JavierMatosD reviewed (dismissed):
Please remove the `schannel` feature since it violates our features as alternatives rule.

JavierMatosD commented on ports/libgit2/portfile.cmake:
```diff
@@ -60,12 +62,18 @@ if(NOT REGEX_BACKEND)
     message(FATAL_ERROR "Must choose pcre or pcre2 regex backend")
 endif()
 
+set(USE_SSH OFF)
+if("openssh" IN_LIST "${FEATURES}")
+    set(USE_SHH "exec")
+elseif("ssh" IN_LIST "${FEATURES}")
+    set(USE_SHH "libssh2") # same as `ON`
+endif()
```
This also looks like an alternative.

razielanarki committed:
removed feature flags

razielanarki committed:
added `STATUS` message about new libgit2 features

as outlined in [maintainer guide](https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#recommended-solutions)

razielanarki committed:
fix cmake `IN_LIST` error
